packages/runtime/container-runtime/src/summaryManager.ts

```diff
@@ -270,9 +209,9 @@ export class SummaryManager extends EventEmitter implements IDisposable {
 
     private refreshSummarizer() {
         // Compute summarizer
-        const newSummarizerClientId = this.quorumHeap.getFirstClientId();
-        if (newSummarizerClientId !== this.summarizerClientId) {
-            this.summarizerClientId = newSummarizerClientId;
+        const newSummarizerClientId = this.orderedClients.getElectedClient()?.clientId;
+        if (newSummarizerClientId !== this.electedClientId) {
+            this.electedClientId = newSummarizerClientId;
```

Comment:
>this.electedClientId [](http://example.com/codeflow?start=12&length=20)

can we get rid of the member this.electedClientId, and just make it a getter that returns this.orderedClients.getElectedClient()?.clientId. I believe this method will only be called when the summarizer has changed. not needed. i just like to reduce state when we can

Reply:
Actually no, I think we should keep it for now. Agree with the sentiment of reducing state, but right now it does prevent excess "summarizer" event emitting. We call this function every time _any_ change happens.